cctp_primitives/src/bit_vector/compression.rs

```diff
@@ -147,6 +147,20 @@ pub fn decompress_bit_vector(compressed_bit_vector: &[u8], expected_size: usize)
     Ok(raw_bit_vector_result)
 }
 
+#[allow(unused_variables)]
+pub fn decompress_bit_vector_without_checks(compressed_bit_vector: &[u8]) -> Result<Vec<u8>, Error> {
```

Comment:
Can you make `decompress_bit_vector` call `decompress_bit_vector_without_checks` ? We avoid some duplicated code

Reply:
Or maybe just using one function and add a boolean flag indicating whether to perform the check or not

Reply:
ok, I will add a single private method with an optional parameter. So both the methods above will use it internally.